new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Kubernetes.java

```diff
@@ -595,6 +595,77 @@ public static boolean isPodRestarted(
     return false;
   }
 
+  /**
+   * Checks if the promethues pods are running in a given namespace.
+   * The method assumes the prometheus pods name to starts with prometheus-server, alertmanager
+   * and decorated with label prometheus
+   * @param namespace in which to check for the pod existence
+   * @return true if pods are exist and running otherwise false
+   * @throws ApiException when there is error in querying the cluster
+   */
+  public static boolean arePrometheusPodsReady(String namespace) throws ApiException {
+    boolean status = false;
+
+    V1Pod pod = getPod(namespace, "component=alertmanager", "prometheus-alertmanager");
+    if (pod != null) {
+      // get the podCondition with the 'Ready' type field
+      V1PodCondition v1PodReadyCondition = pod.getStatus().getConditions().stream()
+          .filter(v1PodCondition -> "Ready".equals(v1PodCondition.getType()))
+          .findAny()
+          .orElse(null);
+
+      if (v1PodReadyCondition != null) {
+        status = v1PodReadyCondition.getStatus().equalsIgnoreCase("true");
+      }
+    } else {
+      logger.info("Prometheus-alertmanager pods don't exist");
+    }
+
+    pod = getPod(namespace, "component=server", "prometheus-server");
+    if (pod != null) {
+      // get the podCondition with the 'Ready' type field
+      V1PodCondition v1PodReadyCondition = pod.getStatus().getConditions().stream()
+          .filter(v1PodCondition -> "Ready".equals(v1PodCondition.getType()))
+          .findAny()
+          .orElse(null);
+
+      if (v1PodReadyCondition != null) {
+        status = v1PodReadyCondition.getStatus().equalsIgnoreCase("true");
+      }
+    } else {
+      logger.info("Prometheus-server pods don't exist");
+    }
+    return status;
+  }
+
+  /**
+   * Checks if the grafana pod is running in a given namespace.
+   * The method assumes the grafana pod name to starts with grafana
+   * and decorated with label grafana
+   * @param namespace in which to check for the pod existence
+   * @return true if pods are exist and running otherwise false
+   * @throws ApiException when there is error in querying the cluster
+   */
+  public static boolean isGrafanaPodReady(String namespace) throws ApiException {
+    boolean status = false;
+
+    V1Pod pod = getPod(namespace, "app.kubernetes.io/name=grafana", "grafana");
+    if (pod != null) {
+      // get the podCondition with the 'Ready' type field
+      V1PodCondition v1PodReadyCondition = pod.getStatus().getConditions().stream()
+          .filter(v1PodCondition -> "Ready".equals(v1PodCondition.getType()))
+          .findAny()
+          .orElse(null);
+
+      if (v1PodReadyCondition != null) {
+        status = v1PodReadyCondition.getStatus().equalsIgnoreCase("true");
+      }
+    } else {
+      logger.info("Grafana pod doesn't exist");
+    }
+    return status;
+  }
+
```

Comment:
can you replace the above 2 methods with this? You can use this method in Grafana,Prometheus

  /**
   * Checks if a pod is ready in a given namespace.
   *
   * @param namespace in which to check if the pod is ready
   * @param labels map of labels as key value pairs
   * @param podName name of the pod to check for
   * @return true if the pod is in the ready condition, false otherwise
   * @throws ApiException if Kubernetes client API call fails
   */
  public static boolean isPodReady(String namespace, Map<String, String> labels, String podName) throws ApiException {
    boolean status = false;
    String labelSelector = null;
    if (labels != null && !labels.isEmpty()) {
      StringBuilder str = new StringBuilder();
      for (Map.Entry<String, String> entry : labels.entrySet()) {
        str.append(entry.getKey()).append("=").append(entry.getValue()).append(",");
      }
      labelSelector = str.toString();
    }

    V1Pod pod = getPod(namespace, labelSelector, podName);
    if (pod != null) {

      // get the podCondition with the 'Ready' type field
      V1PodCondition v1PodReadyCondition = pod.getStatus().getConditions().stream()
          .filter(v1PodCondition -> "Ready".equals(v1PodCondition.getType()))
          .findAny()
          .orElse(null);

      if (v1PodReadyCondition != null) {
        status = v1PodReadyCondition.getStatus().equalsIgnoreCase("true");
        if (status) {
          logger.info("Pod {0} is READY in namespace {1}", podName, namespace);
        }
      }
    } else {
      logger.info("Pod {0} does not exist in namespace {1}", podName, namespace);
    }
    return status;
  }

Reply:
done

Reply:
it may not work, since the other method can accept a null for domainUid and that can cause ambiguity as to which overridden method should be called.

We may have to fix this later. so don't worry about it for now.

Reply:
so I added this for isReadyNginx,     String labelSelector = null;
    return isPodReady(namespace, labelSelector, "nginx-ingress-controller");, so it works. Also both check for null, and execute same code for that